### DISCUSSION TITLE
High CPU usage with Korge

### DISCUSSION THREAD

**Discussion by quanticworld** [korlibs/korge#1396] [General]
Hi,

I'm developping an app with Korge for android mobiles. The CPU usage is very high and the mobile heats easily. I don't have complex animation, only some images display and few event handlers.
Is there a mean to avoid this overhead ? Some best practices ? Have you ever noticed CPU consumption issues ?

Thanks for your help :)

**Comment by soywiz** [comment]
Which device are you using for testing? Are other games using 60fps or 30fps in that phone?

There are CPU consumption issues. Some of them already fixed in alpha versions, other are pending. Also Doubles are using instead of Floats, so the mileage should vary.

Also, are you using debug or release variants? How are you building and launching the app? Can you run the Android profiler too?

**Reply by quanticworld** [reply]
Hi soywiz, thanks for your answer.

I test on my Huawei Mate 10 pro (kirin 970 cpu). Other games can easily run at 60 fps.
Maybe there are some good practices I don't know when working with rendering ? My use case is to load map tiles, have some animated menus, and show a player position on the screen (with geolocation).

I run a debug configuration. I build and launch the app through the launch configuration menu of IntelliJ.
Thanks for your suggestion, I'm going to look into profiler analysis 👀

**Reply by soywiz** [reply]
Thats a high end phone. It should work fine. Have in mind that if you are running in debug the performance can be really poor. Please try running release builds instead

**Reply by soywiz** [reply]
Also if you can provide the versions you are using and information about the profiling, that would help to optimize the parts a that are problematic in your case.
Cherry ok top if you can provide a small snippet or sourcecode reproducing your performance issue

**Reply by quanticworld** [reply]
Hi!

I switched to release build and my phone is cooler, I think this was the main issue I had (debug mode).
When my MVP will be ready, I will profile the app and I'll get back to you if you're interested to look into how your framework behaves :)

(I'm developping an app for outdoor games with geolocation).

A big thanks for KorGE, it is easy to use and have a lot of nice features.
Cheers

**Comment by quanticworld** [comment]
(I close the issue, for now, thanks again)
